services/kinesis/src/it/java/software/amazon/awssdk/services/kinesis/SubscribeToShardIntegrationTest.java

```diff
@@ -162,14 +162,18 @@ public void responseReceived(SubscribeToShardResponse response) {
                                          @Override
                                          public void onEventStream(SdkPublisher<SubscribeToShardEventStream> publisher) {
                                              publisher.limit(3).subscribe(new Subscriber<SubscribeToShardEventStream>() {
+                                                 private Subscription subscription;
                                                  @Override
                                                  public void onSubscribe(Subscription subscription) {
-                                                     subscription.request(10);
+                                                     this.subscription = subscription;
+                                                     subscription.request(1);
                                                  }
 
                                                  @Override
                                                  public void onNext(SubscribeToShardEventStream subscribeToShardEventStream) {
                                                      events.add(subscribeToShardEventStream);
+                                                     //Cancel on first event.
+                                                     subscription.cancel();
                                                  }
 
                                                  @Override
```

Comment:
As discussed offline, we'll remove the `terminalMethodsCalled.set(true)` from `onComplete` because with the latest change, `onComplete` will be invoked. We will also add an assertion to verify `onComplete` is invoked.

In addition, we should add a `Thread.sleep(1000)` before `assertThat(terminalMethodsCalled).isFalse();` because onComplete and complete is called by another thread. Adding the sleep will avoid the race condition.

Reply:
Added new Test case and added new flags completeMethodOfHandlerCalled and completeMethodOfHandlerCalled to avoid confusion.